redis/scan.go

```diff
@@ -27,12 +27,40 @@ var (
 	scannerType = reflect.TypeOf((*Scanner)(nil)).Elem()
 )
 
+// I think this lines that I Add will help you to your bugs
+type ipS struct {
+        Addr addr
+}
+
+type addr struct {
+        net.IP
+}
+
+func (t *addr) RedisScan(src interface{}) error {
+        if t == nil {
+                return errors.New("nil pointer")
+        }
+
+        switch src := src.(type) {
+        case string:
+                return t.UnmarshalText([]byte(src))
+        case []byte:
+                return t.UnmarshalText(src)
+        default:
+                return fmt.Errorf("cannot convert from %T to %T", src, t)
+        }
+}
+
 func ensureLen(d reflect.Value, n int) {
-	if n > d.Cap() {
+	if(n!=nil && d!=nil)
```

Comment:
Unfortunately This in invalid code, `n` is an `int` so can't be nil as such it doesn't compile as can be seen here:
https://go.dev/play/p/H4HgPUpzxPY